src/components/ValidatedInput/ValidFieldText.tsx

```diff
@@ -26,31 +26,39 @@ export class Input extends FieldText {
   }
 }
 
-interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
+export interface ValidFieldTextProps extends React.InputHTMLAttributes<HTMLInputElement> {
   onKeyDown?: (event: React.FormEvent<HTMLInputElement>) => void;
   label: string;
   name: string;
   value: string;
   errMessage?: string;
-  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
+  onChange?: any;
+  isRequired?: boolean;
   optionalLabel?: string;
   isLabelHidden?: boolean;
   type?: "number" | "text" | "password";
   min?: number;
   max?: number;
   helperText?: string;
   tooltip?: string;
-  popoverContent?: React.ReactNode | string;
+  dataQa?: string | null;
+  dataQaProps?: string | null;
   validFieldTextRef?: React.RefObject<HTMLDivElement>;
   rightIcon?: React.ReactNode;
+  isFloat?: boolean;
+  autoComplete?: "on" | "off";
+  showTooltipOnFocus?: boolean;
+  popoverContent?: React.ReactNode;
```

Comment:
AFAIK `React.ReactNode` already contain `undefined` so no need to put `?` flag for `popoverContent`

Reply:
I don't think so, if we remove the `?` mark it will consider the `popoverContent` as a mandatory props, but at the actual implementation not every validfieldtext usage need the popover. And if we put the `popoverContent` as mandatory even though it accept the undefined value typescript will give the validfieldtext warning because it consider as the missing props.